### ISSUE TITLE
Compiler creates incorrect flows in some cases

### ISSUE THREAD

**Issue by nerdoug** [theAgingApprentice/hexbotMQTTScriptCompiler-#28]
In testing out the symbol processing logic, I noticed that the compiled flows don't look right in my test case.

The relevant part of the script looks like this:

> MoveToHomePosition
> CycleStart 1
> MoveRelLastLocal $L, 0, 4, 0,
> Doit $T
> MoveRelLastLocal $l, 4, 0, 0,
> Doit $T
> MoveRelLastLocal $L, 0, 0, -4,
> Doit $T
> MoveRelLastLocal $L, 0, 0, 4,
> Doit $T
> MoveRelHomeLocal $L, -4, 0, 0,
> Doit $T
> MoveRelHomeLocal $L, 0, -4, 0,
> Doit $T
> CycleEnd 1

A snippet from the compiled code looks like this:

> send("Flow,0,MCS,1,0,0,0")
> send("Flow,600,MLC,10,0,0,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-6.6,0 ,13.78,-10.6,0")
> send("Flow,600,MLC,10,0,0,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,17.78,-6.6,0 ,13.78,-10.6,0")
> send("Flow,600,MLC,10,0,0,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,17.78,-6.6,-4 ,13.78,-10.6,0")
> send("Flow,600,MLC,10,0,0,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,17.78,-6.6,0 ,13.78,-10.6,0")
> send("Flow,600,MLC,10,0,0,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,9.78,-10.6,0 ,13.78,-10.6,0")
> send("Flow,600,MLC,10,0,0,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-10.6,0 ,13.78,-14.6,0 ,13.78,-10.6,0")
> send("Flow,0,MCE,1,0,0,0")
> send("Flow,0,DC,1,4,0,0,0")

Extracting the coordinates for leg 5 (which is what $L is defined as):

> ,13.78,    -6.6,     0 
> ,17.78,    -6.6,     0 
> ,17.78,    -6.6,    -4
> ,17.78,    -6.6,     0 
> ,9.78,      -10.6,    0 
> ,13.78,    -14.6,    0

This sequence should have ended up back at the local home position, (13.78, -10.6, 0 ), but Y coordinate (middle one) is wrong.

I don't think it's relevant, but the symbol file looks like this:

> / test symbol template (next line is blank)
> 
> / comment with continuation starting with a space character
>  so I can drone on
>  in extended comments
> 
> /and now we actually define some symbols: 
> / H is the name of the robot, for MQTT purposes
>  Andrew: Hexbot94B97E5F4A40
>  Doug:   Hexbot3C61054ADD98
> H Hexbot3C61054ADD98
> 
> / input file location, including relative path
> I scripts\test-from-folder.txt
> 
> / output file location, including absolute path
> O C:\Users\DougElliott\AppData\Local\MQTT-FX\scripts\01__hb-test-from-folder.js
> 
> / parameterized leg number that will be tested by resulting js code
> L 5
> 
> / number of test cycles to repeat (plus the initial one)
> C 4
> 
> / time interval in milliseconds for each leg movement
> T 600
> 

Needed to capture this case, since I'm going to modify these files in symbol testing. Will debug at next opportunity.

**Comment by nerdoug** [comment]
Doh!  I think this is a simple issue. The last 2 move commands in the cycle use the MoveRelHomeLocal operation. It should have been the MoveRelLastLocal operation, like the rest. When I fixed this and reran it, I got essentially the right numbers, but with a tiny error. Here are the last 2 legs data in the compiled flows:
,13.78,-6.6,0 ,13.78,-10.6,0")
,17.78,-6.6,0 ,13.78,-10.6,0")
,17.78,-6.6,-4 ,13.78,-10.6,0")
,17.78,-6.6,0 ,13.78,-10.6,0")
,13.780000000000001,-6.6,0 ,13.78,-10.6,0")
,13.780000000000001,-10.6,0 ,13.78,-10.6,0")

Looks like some rounding is in order. I think we said that 2 decimal places for coordinates measured in centimeters would be sufficient. I'm surprised that this hasn't shown up earlier. Still investigating.
